<?xml version="1.0" encoding="UTF-8"?>
<rss version="2.0">
<channel>
<title>FHUD</title>
<link>https://nogadaworks.tistory.com/</link>
<description>from caffeine import frappuccino;</description>
<language>ko</language>
<pubDate>Mon, 13 May 2019 06:32:41 +0900</pubDate>
<generator>TISTORY</generator>
<managingEditor>@Jyeo</managingEditor>
<image>
<title>FHUD</title>
<url>https://tistory4.daumcdn.net/tistory/2816583/attach/32cdb9a4ec48486e8974a3aace9737fd</url>
<link>https://nogadaworks.tistory.com</link>
<description>from caffeine import frappuccino;</description>
</image>
<item>
<title>Ubuntu 16.04에서 파이썬 3.5.2 -&gt; 3.7.0으로 업그레이드하기</title>
<link>https://nogadaworks.tistory.com/203</link>
<description>&lt;pre&gt;&lt;code&gt;# 출처: https://serverfault.com/questions/918335/best-way-to-run-python-3-7-on-ubuntu-16-04-which-comes-with-python-3-5
# Install requirements
sudo apt-get install -y build-essential
sudo apt-get install -y checkinstall
sudo apt-get install -y libreadline-gplv2-dev
sudo apt-get install -y libncursesw5-dev
sudo apt-get install -y libssl-dev
sudo apt-get install -y libsqlite3-dev
sudo apt-get install -y tk-dev
sudo apt-get install -y libgdbm-dev
sudo apt-get install -y libc6-dev
sudo apt-get install -y libbz2-dev
sudo apt-get install -y zlib1g-dev
sudo apt-get install -y openssl
sudo apt-get install -y libffi-dev
sudo apt-get install -y python3-dev
sudo apt-get install -y python3-setuptools
sudo apt-get install -y wget

# Prepare to build
mkdir /tmp/Python37
cd /tmp/Python37

# Pull down Python 3.7, build, and install
wget https://www.python.org/ftp/python/3.7.0/Python-3.7.0.tar.xz
tar xvf Python-3.7.0.tar.xz
cd /tmp/Python37/Python-3.7.0
./configure
sudo make altinstall

# 참고: https://stackoverflow.com/questions/41986507/unable-to-set-default-python-version-to-python3-in-ubuntu
update-alternatives --install /usr/bin/python3 python /usr/local/bin/python3.7 10&lt;/code&gt;&lt;/pre&gt;다 하고 나서 다시 버전을 확인하면...&lt;div&gt;&lt;br /&gt;&lt;style type=&quot;text/css&quot;&gt;
p.p1 {margin: 0.0px 0.0px 0.0px 0.0px; font: 14.0px 'Meslo LG S for Powerline'; color: #f4f4f4; background-color: #000000; background-color: rgba(0, 0, 0, 0.85)}
span.s1 {font-variant-ligatures: no-common-ligatures}
&lt;/style&gt;


&lt;p class=&quot;p1&quot;&gt;&lt;span class=&quot;s1&quot;&gt;$ python3 -V&lt;/span&gt;&lt;/p&gt;&lt;p class=&quot;p1&quot;&gt;&lt;span style=&quot;font-variant-ligatures: no-common-ligatures;&quot;&gt;P&lt;/span&gt;&lt;span style=&quot;font-variant-ligatures: no-common-ligatures;&quot;&gt;ython 3.7.0&lt;/span&gt;&lt;/p&gt;&lt;style type=&quot;text/css&quot;&gt;
p.p1 {margin: 0.0px 0.0px 0.0px 0.0px; font: 14.0px 'Meslo LG S for Powerline'; color: #f4f4f4; background-color: #000000; background-color: rgba(0, 0, 0, 0.85)}
span.s1 {font-variant-ligatures: no-common-ligatures}
&lt;/style&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;&lt;p&gt;휴..&lt;/p&gt;&lt;/div&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;</description>
<category>Linux</category>
<author>@Jyeo</author>
<guid>https://nogadaworks.tistory.com/203</guid>
<comments>https://nogadaworks.tistory.com/203#entry203comment</comments>
<pubDate>Fri, 01 Feb 2019 16:16:02 +0900</pubDate>
</item>
<item>
<title>pwndbg 설치 후 pip 실행시 ImportError 해결하기</title>
<link>https://nogadaworks.tistory.com/197</link>
<description>&lt;p&gt;맥북에서 우분투 VM을 돌리고 있는데 스스디가 128짜리라 용량문제(그리고 잘못 설정해서 '에잇! 걍 재설치하자!'하는 문제) 등으로 여러번 삭제 후 환경 구축을 하게 된다.&lt;/p&gt;&lt;p&gt;보통 python(python3은 default로 들어있고 python2가 안 깔려있는데 pwntools를 python3에서 굴리면 좀 귀찮아진다) -&amp;gt; python-pip -&amp;gt; git -&amp;gt; pwndbg(gdb는 기본으로 깔려있다) 순으로&amp;nbsp;설치한다.&amp;nbsp;&lt;/p&gt;&lt;p&gt;그런데 항상 위 순서에 따라서 설치를 진행하면 pwndbg 설치 후 pip를 실행하면 아래 같은 이상한 에러가 뜬다.&lt;/p&gt;&lt;pre&gt;&lt;code&gt;Traceback (most recent call last):
  File &quot;/usr/bin/pip&quot;, line 9, in &lt;module&gt;
    from pip import main
ImportError: cannot import name main
&lt;/module&gt;&lt;/code&gt;&lt;/pre&gt;
&lt;p&gt;pwndbg 설치를 위해서 실행하는 setup.sh에서 pip를 재설치하는 과정이 있던데 이때 path 관련 설정이나 위치가 꼬여서 발생하는 오류인 것 같다.&lt;/p&gt;&lt;pre&gt;&lt;code&gt;sudo python -m pip uninstall pip &amp;amp;&amp;amp; sudo apt install python-pip --reinstall&lt;/code&gt;&lt;/pre&gt;&lt;p&gt;&lt;a href=&quot;https://github.com/pypa/pip/issues/5447#issuecomment-408849802&quot; target=&quot;_blank&quot; class=&quot;tx-link&quot;&gt;https://github.com/pypa/pip/issues/5447#issuecomment-408849802&lt;/a&gt;&lt;br /&gt;&lt;/p&gt;&lt;p&gt;구글링해서 찾아낸 해결책이다(그냥 삭제하고 재설치하면 된다)! 원래는 python3-pip의 솔루션인데 버전만 바꿔서 (우리 상황인) python2에 적용해도 해결된다. 다시 pip를 실행해 보면 제대로 작동하는 것을 확인할 수 있다.&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;</description>
<category>Other</category>
<author>@Jyeo</author>
<guid>https://nogadaworks.tistory.com/197</guid>
<comments>https://nogadaworks.tistory.com/197#entry197comment</comments>
<pubDate>Sun, 12 Aug 2018 19:43:36 +0900</pubDate>
</item>
<item>
<title>Git repo 하나를 다른 repo 아래, 서브디렉토리로 merge하기</title>
<link>https://nogadaworks.tistory.com/196</link>
<description>&lt;p&gt;&lt;a href=&quot;https://stackoverflow.com/questions/1425892/how-do-you-merge-two-git-repositories&quot; target=&quot;_blank&quot; class=&quot;tx-link&quot;&gt;https://stackoverflow.com/questions/1425892/how-do-you-merge-two-git-repositories&lt;/a&gt;&lt;/p&gt;&lt;p&gt;github의 pwnable.kr repo를(폰케알 롸업이 들어있다) 커밋 히스토리를 보존하면서 writeups라는 repo의 아래로&amp;nbsp;보낼 때 사용했다. repo 두 개를 병합한 이후, pwnable.kr는 삭제했다(차피 히스토리가 함께 옮겨졌으므로 잔디밭이 유지된다)&lt;/p&gt;&lt;pre&gt;&lt;code&gt;$ git fetch https://github.com/JunhoYeo/pwnable.kr.git master:pwnkr&lt;/code&gt;&lt;/pre&gt;&lt;p&gt;다른 repo를 잡아먹는(?) 상위 디렉터리가 될 repo(이 경우 writeups)로 이동한 뒤 위 명령어를 사용해 서브디렉토리가 될 repo를 fetch한다.&lt;/p&gt;&lt;pre&gt;$ git merge --allow-unrelated-histories pwnkr
$ ls&lt;code&gt;&lt;/code&gt;&lt;/pre&gt;&lt;p&gt;이후 (현재 디렉토리-writeups-에서 계속) merge하고, 원하는 결과인지 확인한다(ls).&lt;/p&gt;&lt;p&gt;그다음&amp;nbsp;&lt;code&gt;git commit&lt;/code&gt;하고 push하면 끝!&lt;/p&gt;&lt;p&gt;여담으로 ~ 아래에 있는 ./history 파일을 보면 자신이 실행했던 명령어 로그가 남아있으니 이렇게 블로그에 올린다던가 하는 상황에서&amp;nbsp;사용하면 좋더라. 필자는 zsh 쉘을 사용하기 때문에&amp;nbsp;&lt;code&gt;cat ~/.zsh_history&lt;/code&gt;를 이용했다.&lt;/p&gt;</description>
<category>Other</category>
<author>@Jyeo</author>
<guid>https://nogadaworks.tistory.com/196</guid>
<comments>https://nogadaworks.tistory.com/196#entry196comment</comments>
<pubDate>Tue, 07 Aug 2018 18:14:34 +0900</pubDate>
</item>
<item>
<title>Mac에 Python3용 KoNLPy 설치하기</title>
<link>https://nogadaworks.tistory.com/193</link>
<description>&lt;h1 id=&quot;macpython3konlpy&quot;&gt;Mac에 Python3용 KoNLPy 설치하기&lt;/h1&gt;

&lt;p&gt;macOS High Sierra(10.13.3)을 이용하고 있는 필자는 Python3 프로젝트에 한국어 정보처리를 위한 패키지인 KoNLPy를 사용하기로 결정했지만 설치과정에서 엄청난 난관(難關)을 겪는다. 깃허브를 살펴보니 KoNLPy의 이슈가 대부분 Python2 위주로 처리되는 것 같았고, 구글링해도 유용하다고 느껴지는 자료가 많지 않았다. 이것으로 필자가 Python2 EOL 카운트다운을 지켜보는 시간이 늘어날 것 같다. 다른 괜찮은 모듈이나 패키지를 구할 수 없었고(대부분 내부적으로 KoNLPy를 이용해 확장하는 형태가 많았다), 굳이 시대를 역행하면서까지 Python2로 프로젝트를 옮길 이유가 없다고 느껴졌기 때문에 필자는 '반드시 KoNLPy를 설치하는 데 성공하겠노라'는 다짐을 하였고 (to make a long story short) 성공하였다. &lt;/p&gt;

&lt;h2 id=&quot;1konlpy&quot;&gt;1. KoNLPy 설치&lt;/h2&gt;

&lt;pre&gt;&lt;code&gt;$ pip3 install konlpy
Collecting konlpy
  Using cached https://files.pythonhosted.org/packages/b1/41/73127de031d710fa6fc640cc4d4d399977e7a96423131fcd180b9f69627c/konlpy-0.4.4-py2.py3-none-any.whl
Installing collected packages: konlpy
Successfully installed konlpy-0.4.4
&lt;/code&gt;&lt;/pre&gt;

&lt;p&gt;설치는 잘되는데 정작 실행하면 &lt;code&gt;konlpy.tag&lt;/code&gt; -&amp;gt; &lt;code&gt;Twitter&lt;/code&gt; -&amp;gt; &lt;code&gt;Hannanum&lt;/code&gt;에서 &lt;code&gt;jpype&lt;/code&gt;를 import하다가 컥 하고 &lt;code&gt;ModuleNotFoundError: No module named 'jpype'&lt;/code&gt;가 발생한다. &lt;strike&gt;그렇다! 모든 삽질은 &lt;code&gt;jpype&lt;/code&gt;에서 시작되었다. 지금은 잘 해결되었으니 잊혀져라... 잊혀져라...&lt;/strike&gt;&lt;/p&gt;

&lt;h1 id=&quot;2jdk&quot;&gt;2. JDK 설치&lt;/h1&gt;

&lt;p&gt;http://www.oracle.com/technetwork/java/javase/downloads/jdk10-downloads-4416644.html&lt;/p&gt;

&lt;h2 id=&quot;3java_home&quot;&gt;3. &lt;code&gt;JAVA_HOME&lt;/code&gt; 설정&lt;/h2&gt;

&lt;pre&gt;&lt;code&gt;$ /usr/libexec/java_home
/Library/Java/JavaVirtualMachines/jdk-10.0.2.jdk/Contents/Home
$ cat &amp;gt;&amp;gt; ~/.bash_profile
export JAVA_HOME=$(/usr/libexec/java_home)
^C
$ source ~/.bash_profile
$ echo $JAVA_HOME
/Library/Java/JavaVirtualMachines/jdk-10.0.2.jdk/Contents/Home
&lt;/code&gt;&lt;/pre&gt;

&lt;h2 id=&quot;4jpype&quot;&gt;4. JPype 설치&lt;/h2&gt;

&lt;pre&gt;&lt;code&gt;$ pip3 install JPype1-py3
Collecting JPype1-py3
  Using cached https://files.pythonhosted.org/packages/59/90/149647ac2c8649a5983fcc47c78f2881af80cbd99f54248ac31b3d611618/JPype1-py3-0.5.5.2.tar.gz
Building wheels for collected packages: JPype1-py3
  Running setup.py bdist_wheel for JPype1-py3 ... done
  Stored in directory: /Users/junhoyeo/Library/Caches/pip/wheels/76/7d/41/829109c81b0c4bdca8047fb9e0a573d4db3b671fbbe84a84f3
Successfully built JPype1-py3
Installing collected packages: JPype1-py3
Successfully installed JPype1-py3-0.5.5.2
&lt;/code&gt;&lt;/pre&gt;

&lt;h2 id=&quot;5&quot;&gt;5. 성공&lt;/h2&gt;

&lt;p&gt;간단하게 테스트를 해보면 제대로 작동하는 것을 확인할 수 있다. &lt;/p&gt;</description>
<category>Other</category>
<author>@Jyeo</author>
<guid>https://nogadaworks.tistory.com/193</guid>
<comments>https://nogadaworks.tistory.com/193#entry193comment</comments>
<pubDate>Sun, 22 Jul 2018 16:42:53 +0900</pubDate>
</item>
<item>
<title>Python 파일 naming에 주의해야 할 점</title>
<link>https://nogadaworks.tistory.com/162</link>
<description>&lt;p&gt;&lt;a href=&quot;https://tutel.me/c/programming/questions/28697124/facebooksdk+python+module+has+no+attribute+graphapi&quot;&gt;https://tutel.me/c/programming/questions/28697124/facebooksdk+python+module+has+no+attribute+graphapi&lt;/a&gt;&lt;/p&gt;
&lt;p&gt;Python에서 해당 스크립트에서 import하는 모듈 이름과 스크립트 이름을 같게 지정하면 에러가 발생한다.&lt;/p&gt;
&lt;p&gt;인터프리터에서는 되는데 파일을 실행시키면 안 되고... 이거 때문에 몇시간을 헤맸다.&lt;/p&gt;
&lt;p&gt;모듈 이름 역시 해당 모듈의 파일명이므로 충돌이 발생하기 때문이다.&lt;/p&gt;
</description>
<category>Other</category>
<author>@Jyeo</author>
<guid>https://nogadaworks.tistory.com/162</guid>
<comments>https://nogadaworks.tistory.com/162#entry162comment</comments>
<pubDate>Thu, 07 Jun 2018 20:45:26 +0900</pubDate>
</item>
<item>
<title>TensorFlow 설치후 import시의 ImportError 해결</title>
<link>https://nogadaworks.tistory.com/129</link>
<description>&lt;code&gt;
&lt;pre&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;(tensorflow) C:\Users\JunhoYeo&amp;gt;python&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;Python 3.5.5 |Anaconda, Inc.| (default, Mar 12 2018, 17:44:09) [MSC v.1900 64 bit (AMD64)] on win32&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;Type &quot;help&quot;, &quot;copyright&quot;, &quot;credits&quot; or &quot;license&quot; for more information.&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;gt;&amp;gt;&amp;gt; import tensorflow as tf&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;Traceback (most recent call last):&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py&quot;, line 18, in swig_import_helper&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; return importlib.import_module(mname)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\importlib\__init__.py&quot;, line 126, in import_module&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; return _bootstrap._gcd_import(name[level:], package, level)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap&amp;gt;&quot;, line 985, in _gcd_import&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap&amp;gt;&quot;, line 968, in _find_and_load&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap&amp;gt;&quot;, line 957, in _find_and_load_unlocked&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap&amp;gt;&quot;, line 666, in _load_unlocked&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap&amp;gt;&quot;, line 577, in module_from_spec&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap_external&amp;gt;&quot;, line 938, in create_module&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap&amp;gt;&quot;, line 222, in _call_with_frames_removed&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;ImportError: DLL load failed: DLL 초기화 루틴을 실행할 수 없습니다.&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;During handling of the above exception, another exception occurred:&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;Traceback (most recent call last):&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow.py&quot;, line 58, in &amp;lt;module&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; from tensorflow.python.pywrap_tensorflow_internal import *&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py&quot;, line 21, in &amp;lt;module&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; _pywrap_tensorflow_internal = swig_import_helper()&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py&quot;, line 20, in swig_import_helper&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; return importlib.import_module('_pywrap_tensorflow_internal')&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\importlib\__init__.py&quot;, line 126, in import_module&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; return _bootstrap._gcd_import(name[level:], package, level)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;ImportError: No module named '_pywrap_tensorflow_internal'&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;During handling of the above exception, another exception occurred:&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;Traceback (most recent call last):&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;stdin&amp;gt;&quot;, line 1, in &amp;lt;module&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\__init__.py&quot;, line 24, in &amp;lt;module&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; from tensorflow.python import *&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\__init__.py&quot;, line 49, in &amp;lt;module&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; from tensorflow.python import pywrap_tensorflow&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow.py&quot;, line 74, in &amp;lt;module&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; raise ImportError(msg)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;ImportError: Traceback (most recent call last):&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py&quot;, line 18, in swig_import_helper&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; return importlib.import_module(mname)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\importlib\__init__.py&quot;, line 126, in import_module&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; return _bootstrap._gcd_import(name[level:], package, level)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap&amp;gt;&quot;, line 985, in _gcd_import&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap&amp;gt;&quot;, line 968, in _find_and_load&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap&amp;gt;&quot;, line 957, in _find_and_load_unlocked&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap&amp;gt;&quot;, line 666, in _load_unlocked&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap&amp;gt;&quot;, line 577, in module_from_spec&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap_external&amp;gt;&quot;, line 938, in create_module&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;&amp;lt;frozen importlib._bootstrap&amp;gt;&quot;, line 222, in _call_with_frames_removed&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;ImportError: DLL load failed: DLL 초기화 루틴을 실행할 수 없습니다.&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;During handling of the above exception, another exception occurred:&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;Traceback (most recent call last):&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow.py&quot;, line 58, in &amp;lt;module&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; from tensorflow.python.pywrap_tensorflow_internal import *&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py&quot;, line 21, in &amp;lt;module&amp;gt;&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; _pywrap_tensorflow_internal = swig_import_helper()&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py&quot;, line 20, in swig_import_helper&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; return importlib.import_module('_pywrap_tensorflow_internal')&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; File &quot;C:\ProgramData\Anaconda3\envs\tensorflow\lib\importlib\__init__.py&quot;, line 126, in import_module&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;nbsp; &amp;nbsp; return _bootstrap._gcd_import(name[level:], package, level)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;ImportError: No module named '_pywrap_tensorflow_internal'&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;Failed to load the native TensorFlow runtime.&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;See https://www.tensorflow.org/install/install_sources#common_installation_problems&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;for some common reasons and solutions.&amp;nbsp; Include the entire stack trace&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;above this error message when asking for help.&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;&amp;gt;&amp;gt;&amp;gt;&lt;/span&gt;&lt;/p&gt;
&lt;/pre&gt;&lt;/code&gt;
&lt;p&gt;&lt;span style=&quot;font-family: &amp;quot;맑은 고딕&amp;quot;, sans-serif;&quot;&gt;Anaconda에 TensorFlow를 설치하고 import를 시도해 봤더니 위와 같은 에러가 발생했다.&lt;/span&gt;&lt;/p&gt;
&lt;code&gt;&lt;pre&gt;pip install --upgrade https://storage.googleapis.com/tensorflow/windows/cpu/tensorflow-1.0.0-cp35-cp35m-win_amd64.whl
&lt;/pre&gt;&lt;/code&gt;
&lt;p&gt;&lt;a href=&quot;https://stackoverflow.com/questions/42011070/on-windows-running-import-tensorflow-generates-no-module-named-pywrap-tenso&quot; target=&quot;_blank&quot; class=&quot;tx-link&quot;&gt;https://stackoverflow.com/questions/42011070/on-windows-running-import-tensorflow-generates-no-module-named-pywrap-tenso&lt;/a&gt;&lt;/p&gt;&lt;p&gt;위와 같은&amp;nbsp;스택오버플로우에서 찾은 솔루션을 이용하자 아래처럼 정상적으로 사용이 가능했다.&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;&lt;p style=&quot;text-align: center; clear: none; float: none;&quot;&gt;&lt;span class=&quot;imageblock&quot; style=&quot;display:inline-block;width:800px;width: 800px; height: 396px;;height:auto;max-width:100%&quot;&gt;&lt;img src=&quot;https://t1.daumcdn.net/cfile/tistory/9971A23B5AB75A2B05&quot; filemime=&quot;image/jpeg&quot; filename=&quot;fdd.jpg&quot; height=&quot;396&quot; style=&quot;width: 800px; height: 396px;&quot; width=&quot;800&quot;/&gt;&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;
&lt;code&gt;&lt;pre&gt;&lt;p&gt;(tensorflow) C:\Users\JunhoYeo&amp;gt;python&lt;/p&gt;&lt;p&gt;Python 3.5.5 |Anaconda, Inc.| (default, Mar 12 2018, 17:44:09) [MSC v.1900 64 bit (AMD64)] on win32&lt;/p&gt;&lt;p&gt;Type &quot;help&quot;, &quot;copyright&quot;, &quot;credits&quot; or &quot;license&quot; for more information.&lt;/p&gt;&lt;p&gt;&amp;gt;&amp;gt;&amp;gt; import tensorflow as tf&lt;/p&gt;&lt;p&gt;&amp;gt;&amp;gt;&amp;gt;&lt;/p&gt;&lt;/pre&gt;&lt;/code&gt;&lt;p&gt;&lt;br /&gt;&lt;/p&gt;</description>
<category>Develop</category>
<author>@Jyeo</author>
<guid>https://nogadaworks.tistory.com/129</guid>
<comments>https://nogadaworks.tistory.com/129#entry129comment</comments>
<pubDate>Sun, 25 Mar 2018 17:10:02 +0900</pubDate>
</item>
</channel>
</rss>